x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/api.ts

```diff
@@ -33,37 +36,101 @@ import {
 } from './types';
 import { KibanaServices } from '../../../../common/lib/kibana';
 import * as i18n from '../../../pages/detection_engine/rules/translations';
+import { RulesSchema, rulesSchema } from '../../../../../common/detection_engine/schemas/response';
+import { validateEither } from '../../../../../common';
+import {
+  createRulesSchema,
+  updateRulesSchema,
+  patchRulesSchema,
+} from '../../../../../common/detection_engine/schemas/request';
+import { toError, toPromise } from '../../../../../common/shared_imports';
+
+/**
+ * Create provided Rule
+ *
+ * @param rule CreateRulesSchema to add
+ * @param signal to cancel request
+ *
+ * @throws An error if response is not OK
+ */
+const createRule = async ({ rule, signal }: CreateRulesProps): Promise<RulesSchema> =>
+  KibanaServices.get().http.fetch<RulesSchema>(DETECTION_ENGINE_RULES_URL, {
+    method: 'POST',
+    body: JSON.stringify(rule),
+    signal,
+  });
+
+const createRuleWithValidation = async ({ rule, signal }: CreateRulesProps): Promise<RulesSchema> =>
+  pipe(
+    rule,
+    (body) => fromEither(validateEither(createRulesSchema, body)),
+    chain((payload) => tryCatch(() => createRule({ signal, rule: { ...payload } }), toError)),
+    chain((response) => fromEither(validateEither(rulesSchema, response))),
+    flow(toPromise)
+  );
+
+export { createRuleWithValidation as createRule };
 
 /**
- * Add provided Rule
+ * Update provided Rule using PUT
  *
- * @param rule to add
+ * @param rule UpdateRulesSchema to be updated
  * @param signal to cancel request
  *
  * @throws An error if response is not OK
  */
-export const addRule = async ({ rule, signal }: AddRulesProps): Promise<NewRule> =>
-  KibanaServices.get().http.fetch<NewRule>(DETECTION_ENGINE_RULES_URL, {
-    method: rule.id != null ? 'PUT' : 'POST',
```

Comment:
Neither CreateRuleSchema or UpdateRuleSchema require the presence of an `id` so this check was no longer valid.